### PR TITLE
Adding Github Action as CI and add gtest for test.

### PR DESCRIPTION
**Pull request by charsyam**
There are some changes in below:
1] change executable file (program) to libgi.a(library)
 -> I think the main customer use this as a part of their program like library. I think it is better to support this as library to use.
 -> for this, I just add compare.cpp to make this separate with library and test code. 
2] adding gtest for testcases
 -> to address CI/CD, testcases are necessary. GI is c++ project. so there are small candidates. I choose googletest, it is very popular 
3] adding github action as CI.
 -> github supports Gitub Action as CI/CD
 -> whenever there is any code change, It will run the tests in the tests folder.
   but we should add testcases for this.
 -> It uses ubuntu 20.04 LTS as run os.

After merging this PR, you can check the CI with any PR for master.

### CODE REVIEW

File: src/backtrack.cpp [hunk 1]
```diff
@@ -90,7 +97,7 @@ bool Backtrack::run(Coloring* aColoring, Graph* aG1, Graph* aG2, int32_t aNumTre
 	Timer t;
 	t.start();
 	bool result = backtrack(numMatching + numTreeNode);
-	searchTime = t.end();
+	t.end();
```
Comment (NamYehyun) [inline src/backtrack.cpp:100]: Shouldn't the returned value still be assigned to `searchTime` as before?

Comment (daemyungkang) [reply src/backtrack.cpp:100]: Good Point!! @NamYehyun I am moving global variable into each class like searchTime using getSearchTime() method, global variable is not good to handle this library with multi-trheaded.

Comment (NamYehyun) [reply src/backtrack.cpp:100]: I see. I agree `searchTime` shouldn't be a global variable. But, my original question was about the value of `searchTime`. Where is `searchTime` being updated?

Comment (daemyungkang) [reply src/backtrack.cpp]: Hi, @NamYehyun, I fixed to code to set searchTime variable :) Thanks.

Comment (NamYehyun) [reply src/backtrack.cpp]: Thanks to all your works, now our code seems to be better organized. 😃

**Pull request merged**